server/src/services/search.service.ts

```diff
@@ -83,14 +84,27 @@ export class SearchService {
 
     const page = dto.page ?? 1;
     const size = dto.size || 250;
-    const enumToOrder = { [AssetOrder.ASC]: 'ASC', [AssetOrder.DESC]: 'DESC' } as const;
+
+    const enumToOrder = { [AssetOrderPreference.ASC]: 'ASC', [AssetOrderPreference.DESC]: 'DESC' } as const;
+
+    const mapOrder = (order?: AssetOrder) => {
+      if (order) {
+        if (order === AssetOrder.PREFERENCE) {
+          return enumToOrder[auth.user.preferedAlbumOrder];
+        }
```

Comment:
This does not need to be in the outer if. You can simply move it in front of it :)